[SALEOR-6404] Webhooks without secretKey are also signed (#435)
saleor/saleor-docs: docs/developer/extending/apps/asynchronous-webhooks.mdx

All webhooks with `targetUrl`, where the scheme is HTTP(S) will use this protocol.

Saleor calculates a payload signature under `Saleor-Signature`:

>  * if `secretKey` was set for a webhook - the HMAC sha256 header based on it and the payload

Reviewer: ```suggestion
 * if `secretKey` was set for a webhook - the HMAC SHA-256 header based on it and the payload
```

>  * if `secretKey` was set for a webhook - the HMAC SHA-256 header based on it and the payload

Reviewer 2: I think that this note is related to async and sync webhooks.
Also, don't you think, that it would be good to add here a small note about the place where you can find a public key? Something like this: https://docs.saleor.io/docs/3.x/developer/users#rs256-public-key

>  * if `secretKey` wasn't set for a webhook - the RS256 signature based on our private key used also to sign JTW tokens and the payload (available since Saleor 3.5, before payload from webhooks without `secretKey` set were not signed)

Reviewer: ```suggestion
 * if `secretKey` wasn't set for a webhook - the RS256 signature based on our private key used also to sign JWT tokens and the payload (available since Saleor 3.5, before payload from webhooks without `secretKey` set were not signed)
```

During HTTPS requests there are several headers included:
- Saleor-Event - defines event which is assigned to the webhook
- Saleor-Domain - defines saleor domain

> - Saleor-Signature - defines signature to indicate that the request is veriferable

Reviewer: ```suggestion
- Saleor-Signature - defines signature to indicate that the request is verifiable
```
